crates/catalog/inmemory/src/catalog.rs

```diff
@@ -0,0 +1,1511 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! This module contains in-memory catalog implementation.
+
+use async_lock::Mutex;
```

Comment:
Hi, is there a specific reason we're using `async_lock`? Could we switch to `futures::lock::Mutex` since we already depend on `futures`?

Reply:
I wasn't aware of `futures`/`futures::lock::Mutex` 🙂 

The only difference I can see between the two is that `async_lock::Mutex` docs state that: 
```
The locking mechanism uses eventual fairness to ensure locking will be fair on average without sacrificing performance. This is done by forcing a fair lock whenever a lock operation is starved for longer than 0.5 milliseconds.
```

Whereas `futures::lock::Mutex` docs state that: 
```
This mutex provides no fairness guarantees. Tasks may not acquire the mutex in the order that they requested the lock, and it’s possible for a single task which repeatedly takes the lock to starve other tasks, which may be left waiting indefinitely.
```

Regardless, I've switched to `futures::lock::Mutex` since I don't think this should be an issue (and we can always re-evaluate in the future if it does become one).